include/dlaf/matrix/matrix.h

```diff
@@ -128,6 +128,14 @@ class Matrix : public Matrix<const T, device> {
     return operator()(this->distribution().localTileIndex(index));
   }
 
+  auto readwrite_sender(const LocalTileIndex& index) noexcept {
+    return this->operator()(index);
```

Comment:
A `future` is also a sender. The default behaviour of `future` and `shared_future` as senders is that they will send (call `set_value` on a receiver) the contained value, not the whole future (like `future::then` would do). For `future` the default behaviour is ok since the value is not shared and we can just use it directly as a sender. For `shared_future` we have to "keep" the future to handle the lifetime correctly. This does mean though that at the moment the `transform`s that have been added also do an `unwrapping` to get a reference to the wrapped value from a `shared_future`. This dance is highly related to my earlier comments about lifetime management being tied to futures and I'd like to change that. However, I still want to keep futures as the senders for the moment to avoid having to change all of DLAF at once.